pkg/apis/io/v1alpha1/jaeger_types.go

```diff
@@ -150,6 +151,14 @@ type JaegerDependenciesSpec struct {
 	ElasticsearchNodesWanOnly   bool   `json:"elasticsearchNodesWanOnly"`
 }
 
+// JaegerEsIndexCleanerSpec holds the options related to es-index-cleaner
+type JaegerEsIndexCleanerSpec struct {
+	Enabled      bool   `json:"enabled"`
```

Comment:
By default disabled. 

I think we should enable it by default along with dependencies job. I would rather do it in a separate PR to keep this minimal.